tests/library/browsercontext-credentials.spec.ts

```diff
@@ -76,3 +77,64 @@ it('should return resource body', async ({ browser, server }) => {
   expect((await response.body()).toString()).toContain('Playground');
   await context.close();
 });
+
+it('should work with correct credentials and matching origin', async ({ browser, server }) => {
+  server.setAuth('/empty.html', 'user', 'pass');
+  const origin = server.PREFIX;
+  const context = await browser.newContext({
+    httpCredentials: { username: 'user', password: 'pass', origin: origin }
+  });
+  const page = await context.newPage();
+  const response = await page.goto(server.EMPTY_PAGE);
+  expect(response.status()).toBe(200);
+  await context.close();
+});
+
+it('should work with correct credentials and matching origin case insensitive', async ({ browser, server }) => {
+  server.setAuth('/empty.html', 'user', 'pass');
+  const origin = server.PREFIX.toUpperCase();
+  const context = await browser.newContext({
+    httpCredentials: { username: 'user', password: 'pass', origin: origin }
+  });
+  const page = await context.newPage();
+  const response = await page.goto(server.EMPTY_PAGE);
+  expect(response.status()).toBe(200);
+  await context.close();
+});
+
+it('should fail with correct credentials and mismatching scheme', async ({ browser, server }) => {
+  server.setAuth('/empty.html', 'user', 'pass');
+  const origin = server.PREFIX.includes('http://') ? server.PREFIX.replace('http://', 'https://') : server.PREFIX.replace('https://', 'http://');
+  const context = await browser.newContext({
+    httpCredentials: { username: 'user', password: 'pass', origin: origin }
```

Comment:
```suggestion
    httpCredentials: { username: 'user', password: 'pass', origin: server.PREFIX.replace('http://', 'https://') }
```
server has always `http:` protocol. Drop again the `origin` variable.